app/policies/user_policy.rb

```diff
@@ -1,7 +1,22 @@
 class UserPolicy < ApplicationPolicy
-  class Scope < Scope
-    def resolve
-      scope
-    end
+  def initialize(user, record_user)
+    @user = user
+    @record_user = record_user
   end
+
+  def show?
+    user.admin? || user == record_user
+  end
```

Comment:
I like the way you phrased this to give admins permission for all users, and regular users permission for their own account.